rasa/shared/exceptions.py

```diff
@@ -69,3 +69,7 @@ class FileNotFoundException(RasaException, FileNotFoundError):
 
 class FileIOException(RasaException):
     """Raised if there is an error while doing file IO."""
+
+
+class InvalidConfigException(ValueError, RasaException):
```

Comment:
for my understanding: Why did we change the suffix from `Error` to `Exception`? Configuration errors are usually something we can't recover from - so `Error` seems more suited in my opinion (following [this](https://www.geeksforgeeks.org/errors-v-s-exceptions-in-java/) or [this](https://www.datacamp.com/community/tutorials/exception-handling-python) guide.

Reply:
I agree with you, it should be an `Error`. I was thinking about this as well and thought that we have quite a number of `Exceptions` that should rather be `Errors` by that definition (e.g. the above yaml errors) that it might be more consistent now to just always use `Exception`. I am also not sure if this is always a useful distinction. E.g. I certainly agree that OOM is something different than a model not found exception, but things start to break when you look at the `ValueError` which in some cases should be treated as an Exception and in some places more like an Error. It assumes that the throwing function knows what the caller is able to handle and recover from, which I am not sure is the case (e.g. why shouldn't there be a function that is able to recover from an `InvalidConfigException` by "repairing" a common error).

Long story short rather than using a differentiation where I am not sure if it provides any value I aimed to keep the naming consistent between the exceptions we raise. But honestly, this is a tossup and happy to change it to `Error`.

Reply:
> It assumes that the throwing function knows what the caller is able to handle and recover from, which I am not sure is the case (e.g. why shouldn't there be a function that is able to recover from an InvalidConfigException by "repairing" a common error)

I agree 👍 

> Long story short rather than using a differentiation where I am not sure if it provides any value I aimed to keep the naming consistent between the exceptions we raise

Makes sense 👍 Just wanted to know the reason since we recently named a couple of exceptions / errors in Rasa X `XyError` .